### PR TITLE
azure: add udev rules needed by k8s cloudprovider

### LINKED ISSUE #1614: azure: PVC/PV disk mount does not work
## What keywords did you search in tectonic-installer issues before filing this one?

`azure`

## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

Deploy an Azure cluster. Deploy a Managed Disk "Fast" Storage Class from the examples (https://github.com/khenidak/kubernetes-azure-managed-disks).

Then provision a naive PVC and wait for the PV to be created. Now launch a Pod that consumes that PV/PVC. It will fail to start due to this upstream issue: https://github.com/kubernetes/kubernetes/issues/50150

## Versions

* Tectonic version ([release](https://github.com/coreos/tectonic-installer/releases) or commit hash): HEAD of master as of filing this ()
* Terraform version (`terraform version`): n/a
* Platform (aws|azure|openstack|metal): Azure

## What happened?

Created an Azure cluster and tried to deploy an auto-provisioned PV/PVC and a Pod that consumes it. Instead, the Pod fails to start because the mounting process fails. Again, please see the upstream issue mentioned just a few lines above.

## What you expected to happen?

The PV to be created, attached, mounted, and the Pod starts.

## How to reproduce it (as minimally and precisely as possible)?

See above.

## Anything else we need to know?

N/A

### PR DESCRIPTION
**Pull request by colemickens**
Fixes #1614 by adding udev rules needed by the Azure cloudprovider. See this comment for context: https://github.com/kubernetes/kubernetes/issues/50150#issuecomment-320842952

### CODE REVIEW

**Comment by colemickens** [comment]
I'm not sure if it would be more appropriate to add this to Container Linux itself, but this seems like a potentially desirable fix in the case of expediency.

**Comment by colemickens** [comment]
I'm not sure if it would be more appropriate to add this to Container Linux itself, but this seems like a potentially desirable fix in the case of expediency.

Also, this was tested with: https://gist.github.com/colemickens/0cd6a12fdbf5a8b5040724a0b8e900fd.

**Comment by squat** [comment]
@colemickens, thanks for contributing this. In order to reduce code duplication, could you implement this in a module following the pattern that the tx-off.service uses?

File: modules/azure/master-as/resources/66-azure-storage.rules [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ENV{DEVTYPE}=="partition", PROGRAM="/bin/sh -c 'readlink /sys/class/block/%k/../device|cut -d: -f4'", ENV{fabric_name}="$env{fabric_scsi_controller}/lun$result", GOTO="azure_names"
+PROGRAM="/bin/sh -c 'readlink /sys/class/block/%k/device|cut -d: -f4'", ENV{fabric_name}="$env{fabric_scsi_controller}/lun$result", GOTO="azure_names"
+GOTO="azure_end"
+
+# Create the symlinks
+LABEL="azure_names"
+ENV{DEVTYPE}=="disk", SYMLINK+="disk/azure/$env{fabric_name}"
+ENV{DEVTYPE}=="partition", SYMLINK+="disk/azure/$env{fabric_name}-part%n"
+
+LABEL="azure_end"
```
Comment (squat) [inline modules/azure/master-as/resources/66-azure-storage.rules]: This should probably have a newline

**Comment by colemickens** [comment]
@squat Both requested changes have been made. Let me know and I can squash.

**Comment by alexsomesan** [comment]
@colemickens I'm giving this a quick manual test and will then merge if everything works fine.

File: modules/azure/worker-as/resources/66-azure-storage.rules [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ENV{DEVTYPE}=="partition", PROGRAM="/bin/sh -c 'readlink /sys/class/block/%k/../device|cut -d: -f4'", ENV{fabric_name}="$env{fabric_scsi_controller}/lun$result", GOTO="azure_names"
+PROGRAM="/bin/sh -c 'readlink /sys/class/block/%k/device|cut -d: -f4'", ENV{fabric_name}="$env{fabric_scsi_controller}/lun$result", GOTO="azure_names"
+GOTO="azure_end"
+
+# Create the symlinks
+LABEL="azure_names"
+ENV{DEVTYPE}=="disk", SYMLINK+="disk/azure/$env{fabric_name}"
+ENV{DEVTYPE}=="partition", SYMLINK+="disk/azure/$env{fabric_name}-part%n"
+
+LABEL="azure_end"
```
Comment (Quentin-M) [inline modules/azure/worker-as/resources/66-azure-storage.rules]: no new line

**Comment by alexsomesan** [comment]
I was able to confirm this is working by spinning up a cluster from this branch.
I used the manifests in the gist provided by @colemickens.
```
 alex@Alexs-MBP  ~/go/src/github.com/coreos/tectonic-installer   azure-udev  kubectl get pvc
NAME                 STATUS    VOLUME                                     CAPACITY   ACCESSMODES   STORAGECLASS   AGE
dd-managed-fast-01   Bound     pvc-1da381a2-7c65-11e7-978b-000d3a2a4b59   128Gi      RWO           managedfast    8m
 alex@Alexs-MBP  ~/go/src/github.com/coreos/tectonic-installer   azure-udev  kubectl get pv
NAME                                       CAPACITY   ACCESSMODES   RECLAIMPOLICY   STATUS    CLAIM                        STORAGECLASS   REASON    AGE
pvc-1da381a2-7c65-11e7-978b-000d3a2a4b59   128Gi      RWO           Delete          Bound     default/dd-managed-fast-01   managedfast              7m
 alex@Alexs-MBP  ~/go/src/github.com/coreos/tectonic-installer   azure-udev  kubectl get pods
NAME                              READY     STATUS    RESTARTS   AGE
test-azure-disk-396068938-kdbnh   2/2       Running   0          8m
```

**Comment by colemickens** [comment]
@philips In the latest 16.04 images (in Azure at least), these udev rules are included. On the upstream issue I've commented that this issue is likely to affect other distros and older ubuntu images.

**Pull request merged**